### DISCUSSION TITLE
I cannot resize component off canvas with default config

### DISCUSSION THREAD

**Discussion by ngoc199** [GrapesJS/grapesjs#4981] [Q&A] [answered]
I want to create an image component that can be resized as big as the user wants, which can be off the canvas.

Here is the code to update the component of `grapesjs`:

```ts
  updateRect(store: boolean) {
    const el = this.el!;
    const resizer = this;
    const config = this.opts;
    const rect = this.rectDim!;
    const updateTarget = this.updateTarget;
    const selectedHandler = this.getSelectedHandler();
    const { unitHeight, unitWidth, keyWidth, keyHeight } = config;

    // Use custom updating strategy if requested
    if (isFunction(updateTarget)) {
      updateTarget(el, rect, {
        store,
        selectedHandler,
        resizer,
        config,
      });
    } else {
      const elStyle = el.style as Record<string, any>;
      elStyle[keyWidth!] = rect.w + unitWidth!;
      elStyle[keyHeight!] = rect.h + unitHeight!;
    }

    this.updateContainer();
  }
```

With the default config, which `updateTarget` is `undefined` by default, the component could be resized off canvas.

However, the size of the component is limited by the size of the canvas.

Then I came to the pull request [#1736](https://github.com/GrapesJS/grapesjs/pull/1736). In that code, the `updateTarget` function limits the size of the component to the size of the canvas. Though it is the old code, it let me know I have to create my `updateTarget` function to do what I want.

Here is my `updateTarget` function:

```ts
  updateTarget: (el: HTMLElement, rect: RectDim, opts: any) => {
    const { keyWidth, keyHeight, unitHeight, unitWidth } = opts.config;
    console.log(opts)
    el.style[keyWidth] = `${rect.w}${unitWidth}`
    el.style[keyHeight] = `${rect.h}${unitHeight}`
  },
```

After adding this, the component can be resized off canvas.

Although my problem is solved, I cannot find the code that limits the size of the component in this repository.

Could you show me that code? I want to understand this problem better.

**Comment by artf** [comment, marked as answer]
> Although my problem is solved, I cannot find the code that limits the size of the component in this repository.

The code which limits the size is still in the same [command](https://github.com/GrapesJS/grapesjs/blob/a2ca4b1a93520e61cc7e49a42f103d5887e33cf2/src/commands/view/SelectComponent.ts#L448-L452)
